components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java

```diff
@@ -49,8 +49,22 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
             settingsDTO.setApplicationSharingEnabled(APIUtil.isMultiGroupAppSharingEnabled());
             settingsDTO.setMapExistingAuthApps(APIUtil.isMapExistingAuthAppsEnabled());
             Map<String, Environment> environments = APIUtil.getEnvironments();
-            Environment environment = environments.get(RestApiConstants.DEFAULT_ENVIRONMENT);
-            settingsDTO.apiGatewayEndpoint(environment.getApiGatewayEndpoint());
+            if (environments.size()==1){
+                Map.Entry<String,Environment> entry = environments.entrySet().iterator().next();
+                Environment singleEnvironment = environments.get(entry.getKey());
+                settingsDTO.apiGatewayEndpoint(singleEnvironment.getApiGatewayEndpoint());
+            } else {
```

Comment:
I don't think we need if-else here. Else logic can handle single GW scenario too right?

Reply:
Fixed with abae412